File: src/main/java/me/dolphago/controller/FollowController.java
package me.dolphago.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.sun.istack.NotNull;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import me.dolphago.dto.ResponseDto;
import me.dolphago.service.FollowTrackingService;

@RequiredArgsConstructor
@RestController
@Slf4j
public class FollowController {
    private final FollowTrackingService followTrackingService;

    @GetMapping("/check")
    public ResponseEntity<ResponseDto> check(@NotNull @RequestParam("handle") String handle) {
        log.info("Check Follow {}'s Status.....", handle);
        return ResponseEntity.ok(followTrackingService.checkFollow(handle));
    }
}

File: src/main/kotlin/me/dolphago/controller/FollowController.kt
package me.dolphago.controller

import io.github.oshai.kotlinlogging.KotlinLogging
import me.dolphago.dto.ResponseDto
import me.dolphago.service.FollowTrackingService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class FollowController(
    val followTrackingService: FollowTrackingService
) {
    val log = KotlinLogging.logger { }

    @GetMapping("/check")
    fun check(@RequestParam("handle") handle: String): ResponseEntity<ResponseDto> {
        log.info { "Check Follow ${handle}'s Status....." }
        return ResponseEntity.ok(followTrackingService.checkFollow(handle))
    }
}
